discovery-provider/src/api/v1/tracks.py

```diff
@@ -534,19 +596,41 @@ def get(self, version):
         return success_response(recommended_tracks[:limit])
 
 
+@ns.route("/recommended/<string:version>")
+class RecommendedTrackWithVersion(RecommendedTrack):
+    @record_metrics
+    @ns.doc(
+        id="""Get Recommended Tracks With Version""",
+        description="""Get recommended tracks using the given trending strategy version""",
+        params={"version": "The strategy version of trending to use"},
+    )
+    @ns.expect(recommended_track_parser)
+    @ns.marshal_with(tracks_response)
+    @cache(ttl_sec=TRENDING_TTL_SEC)
+    def get(self, version):
+        super().get(self, version)
+
+
 full_recommended_track_parser = recommended_track_parser.copy()
-full_recommended_track_parser.add_argument("user_id", required=False)
+full_recommended_track_parser.add_argument(
+    "user_id", required=False, description="The user ID of the user making the request"
+)
 
 
 @full_ns.route(
     "/recommended",
     defaults={"version": DEFAULT_TRENDING_VERSIONS[TrendingType.TRACKS].name},
     strict_slashes=False,
 )
-@full_ns.route("/recommended/<string:version>")
-class FullRecommendedTrack(Resource):
+class FullRecommendedTracks(Resource):
```

Comment:
I don't think we want to drop the ability to specify the version - it should help with migrations of default trending versions

Reply:
I added `FullRecommendedTracksWithVersion` to replace this route here which the same thing. It needs to be separate so that they can have different operation IDs (we can't have two routes with the same ID)